avm/src/vm/opcode.rs

```diff
@@ -2992,32 +3013,33 @@ pub static OPCODES: Lazy<HashMap<i64, ByteOpcode>> = Lazy::new(|| {
         ));
       }
     };
+    // TODO: get rid of the potential panic here
     let body_str = str::from_utf8(&body_req).unwrap().to_string();
     let body = HandlerMemory::str_to_fractal(&body_str);
     // Populate the event and emit it
-    event.init_fractal(0);
-    event.push_fractal(0, method);
-    event.push_fractal(0, url);
+    event.init_fractal(0)?;
+    event.push_fractal(0, method)?;
+    event.push_fractal(0, url)?;
     HandlerMemory::transfer(
       &headers_hm,
       CLOSURE_ARG_MEM_START,
       &mut event,
       CLOSURE_ARG_MEM_START,
-    );
-    event.push_register(0, CLOSURE_ARG_MEM_START);
-    event.push_fractal(0, body);
+    )?;
+    event.push_register(0, CLOSURE_ARG_MEM_START)?;
+    event.push_fractal(0, body)?;
     // Generate a threadsafe raw ptr to the tx of a watch channel
     // A ptr is unsafely created from the raw ptr in httpsend once the
     // user's code has completed and sends the new HandlerMemory so we
     // can resume execution of this HTTP request
     let (tx, rx): (Sender<Arc<HandlerMemory>>, Receiver<Arc<HandlerMemory>>) = oneshot::channel();
     let tx_ptr = Box::into_raw(Box::new(tx)) as i64;
-    event.push_fixed(0, tx_ptr);
+    event.push_fixed(0, tx_ptr)?;
     let event_emit = EventEmit {
       id: i64::from(BuiltInEvents::HTTPCONN),
       payload: Some(event),
     };
-    let event_tx = EVENT_TX.get().unwrap();
+    let event_tx = EVENT_TX.get().ok_or(VMError::ShutDown)?;
```

Comment:
Hmmm interesting, does this work as `unwrap_or`?

Reply:
no, since `ok_or` still returns the `Result`, whereas `unwrap_or` is the same as `getOr` in Alan (if it's an `Err`, returns an `Ok` with the given default value)

Reply:
Got it